static/app/components/performance/teamKeyTransaction.tsx

```diff
@@ -1,416 +1,132 @@
-  isOpen: boolean;
-  keyedTeams: Team[] | null;
-  disabled?: boolean;
-  initialValue?: number;
-};
-
-type Props = {
+type TeamKeyTransactionProps = Omit<
+  MultipleSelectProps<string>,
+  'multiple' | 'options' | 'value' | 'defaultValue' | 'onChange' | 'title'
+> & {
   counts: Map<string, number> | null;
-  error: string | null;
   handleToggleKeyTransaction: (selection: TeamSelection) => void;
-  isLoading: boolean;
   keyedTeams: Set<string> | null;
   organization: Organization;
   project: Project;
   teams: Team[];
-  title: React.ComponentType<TitleProps>;
   transactionName: string;
-  initialValue?: number;
 };
 
-type State = {
-  isOpen: boolean;
-};
-
-class TeamKeyTransaction extends Component<Props, State> {
-  state: State = {
-    isOpen: false,
-  };
-
-  componentDidUpdate(_props: Props, prevState: State) {
-    if (this.state.isOpen && prevState.isOpen === false) {
-      document.addEventListener('click', this.handleClickOutside, true);
-    }
-    if (this.state.isOpen === false && prevState.isOpen) {
-      document.removeEventListener('click', this.handleClickOutside, true);
-    }
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('click', this.handleClickOutside, true);
-  }
-
-  private menuEl: Element | null = null;
-
-  handleClickOutside = (event: MouseEvent) => {
-    if (!this.menuEl) {
-      return;
-    }
-    if (!(event.target instanceof Element)) {
-      return;
-    }
-    if (this.menuEl.contains(event.target)) {
-      return;
-    }
-    this.setState({isOpen: false});
-  };
-
-  toggleOpen = () => {
-    this.setState(({isOpen}) => ({isOpen: !isOpen}));
-  };
-
-  toggleSelection = (enabled: boolean, selection: TeamSelection) => () => {
-    const {handleToggleKeyTransaction, organization} = this.props;
-    const {action} = selection;
-    trackAnalytics('performance_views.team_key_transaction.set', {
-      organization,
-      action,
-    });
-
-    return enabled ? handleToggleKeyTransaction(selection) : undefined;
-  };
-
-  partitionTeams(counts: Map<string, number>, keyedTeams: Set<string>) {
-    const {teams, project} = this.props;
+function TeamKeyTransaction({
+  keyedTeams,
+  teams,
+  project,
+  counts,
+  handleToggleKeyTransaction,
+  transactionName,
+  organization,
+  ...props
+}: TeamKeyTransactionProps) {
+  const projectTeams = useMemo(() => new Set(project.teams.map(({id}) => id)), [project]);
+
+  const value = useMemo(
+    () => (keyedTeams ? [...projectTeams].filter(teamId => keyedTeams.has(teamId)) : []),
+    [keyedTeams, projectTeams]
+  );
 
+  const options = useMemo<MultipleSelectProps<string>['options']>(() => {
     const enabledTeams: Team[] = [];
     const disabledTeams: Team[] = [];
-    const noAccessTeams: Team[] = [];
 
-    const projectTeams = new Set(project.teams.map(({id}) => id));
     for (const team of teams) {
       if (!projectTeams.has(team.id)) {
-        noAccessTeams.push(team);
```

Comment:
@doralchan and I both agree that we don't need to include "no-access" teams in the menu
<img width="220" alt="image" src="https://github.com/getsentry/sentry/assets/44172267/f506dc2b-86ea-47c2-8727-c938cdfce14a">

Despite its name, the "no-access" section doesn't contain teams that the user doesn't have access to. Rather, it's the list of teams that aren't associated with the project that the highlighted transaction is in.

For example, the following transaction is in the `javascript` project. That project is associated with the `sentry` and `support-test` teams. The `new-page-filters` team is not associated with `javascript`, so it's in the "no-access" section.
<img width="718" alt="image" src="https://github.com/getsentry/sentry/assets/44172267/9e9786b9-0abb-4aa6-b7af-16d0a5a6cae8">

There doesn't seem to be a good reason for `new-page-filters` to be included. If it was a matter of user access, then displaying the team alongside a link to request access would make sense. But that's not the case here.